---
title: "Single Cell Transcriptomics Activity"
output: html_document
---

Note the galaxy link they should use: https://usegalaxy.org/join-training/itn-at-moffitt-2025/

## Tutorial Steps

We will be working through a modified/truncated version of the [Galaxy Training Tutorial "Filter, plot, and explore single cell RNA-seq data with Seurat" Tutorial ](https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html). 

Changes include:

  - several subsections within QC will be omitted
  - we rearranged the Dimensionality Reduction section
  - we will stop before the Differential Expression Testing: Finding Markers section

### Open Galaxy

<input type="checkbox"> Please [use this link to access the Galaxy platform specifically for our activity](https://usegalaxy.org/join-training/itn-at-moffitt-2025/).<br/>

Note: this link will only be active for this workshop, but you can continue to use Galaxy resources for free by visiting [usegalaxy.org](https://usegalaxy.org/).

### Set up Galaxy's history pane

<input type="checkbox"> If you have files in your history already, use the `+` button on the top right of the history pane to `Create new history`. <br/>
<input type="checkbox"> Click the pencil button next to "Unnamed history". Fill in the name with something descriptive/appropriate. Click "Save".

<p align="center" >
    <img width="40%" src="resources/images/hist_singlecell.png" class="img_shad">
</p>

Now we are ready to start the tutorial!

### Open the tutorial

<input type="checkbox"> Click on the [full tutorial link](https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html) to follow along.

### Get Data onto Galaxy and generate a Seurat object

<input type="checkbox"> Click on the "Get Data onto Galaxy and generate a Seurat object" link to get started in the tutorial.

### 

#### EBI Data Retrieval

<input type="checkbox"> Copy and paste  `EBI SCXA Data Retrieval` into the tool search.

<p align="center" >
    <img width="40%" src="resources/images/data_singlecell.png" class="img_shad">
</p>

<input type="checkbox"> Select `EBI SCXA DATa Retrieval`.

<input type="checkbox"> Copy paste `E-MTAB-6945` into the `SC-Atlas experiment accession` field 

<input type="checkbox"> Make sure the `“Choose the type of matrix to download”` field says `Raw filtered counts`.

<input type="checkbox"> Click the Run tool button.

<p align="center" >
    <img width="50%" src="resources/images/EBI_singlecell.png" class="img_shad">
</p>

You will see your history pane show some new activity, with rectangles showing the data file upload first in gray (to show that the job has been requested), then yellow (to show that Galaxy is working on it), and finally green (to show that the job is complete). 

Note: do not worry about the numbers next to each step - depending on how many times you run requests you will have different numbers. 

<p align="center" >
    <img width="50%" src="resources/images/gal_data_singlecell.png" class="img_shad">
</p>


You have now imported the `exp_design.tsv`, `barcodes.tsv`, `genes.tsv`, `matrix.mtx` and files into Galaxy! Note that each file starts with `EBI SCXA Data Retrieval on E-MTAB-6945` to indicate how we got them. 

#### Generating a Seurat Read10X Object

Now we need to get our data into a format that the Seurat tools will recognize. 

<input type="checkbox"> Copy paste `Seurat Read10x` into the tools search.

<p align="center" >
    <img width="50%" src="resources/images/10x_singlecell.png" class="img_shad">
</p>

<input type="checkbox"> Select `Seurat Read10x`.

<input type="checkbox"> Select the pull down menu for the `Expression matrix in sparse matrix format (.mtx)` and select the `matrix.mtx` file.

<p align="center" >
    <img width="50%" src="resources/images/select_file_singlecell.png" class="img_shad">
</p>

Continue with the rest of the files to select the appropriate file for each field. Remeber that each file starts with `EBI SCXA Data Retrieval on E-MTAB-6945`.

<input type="checkbox"> Select the pull down menu for the `Gene table` and select the `genes.tsv` file.

<input type="checkbox"> Select the pull down menu for the `Barcode/cell table` and select the `barcodes.tsv` file.

<input type="checkbox"> Select the pull down menu for the `Cell Metadata` and select the `exp_design.tsv` file.

<input type="checkbox"> Type `5` into the `Minimum cells to include features`.

<input type="checkbox"> Ensure that the `Choose format of the output` field says `RDS with a Seurat object`.

<input type="checkbox"> Rename the output to `Initial Seurat Object`.

<p align="center" >
    <img width="50%" src="resources/images/seurat_singlecell.png" class="img_shad">
</p>


<input type="checkbox"> Click `Run Tool` Button.

You should see a green message indicating that the job is now in the queue. 

<input type="checkbox"> Click on the pencil button of the new output that shows up in the history pane.

<p align="center" >
    <img width="50%" src="resources/images/rename_singlecell.png" class="img_shad">
</p>

<input type="checkbox"> Replace the name of the object to `Initial Seurat Object` and press save.

<p align="center" >
    <img width="50%" src="resources/images/rename2_singlecell.png" class="img_shad">
</p>


Now we have a Seurat object so we can start using Seurat to work with our data!

### QC and further processing of the Seurat object

Our first step is to do some Quality Control checks aka QC to evaluate our data.

#### QC: Visualize Counts

There will be a number of ‘cells’ that are actually just empty droplets or low-quality. There will also be genes that could be sequencing artifacts or that appear with such low frequency that statistical tools will fail to accurately analyse them.

This background noise of both cells and genes not only makes it harder to distinguish real biological information from artifacts, but also makes it computationally demanding to analyze.

We want to filter our cells, but first we need to know what our data looks like to make appropriate filters.

We will not have time to go through all of the possible plots one might use, but we will do some evaluations. Remember that you can always do the full tutorial in your own time.

<input type="checkbox"> Search for the `Plot with Seurat` tool in the tool search. 

<input type="checkbox"> Select `**Plot** with Seurat`. 

<p align="center" >
    <img width="50%" src="resources/images/plot_tool_singlecell.png" class="img_shad">
</p>

<input type="checkbox"> Ensure that the RDS file field shows our Initial Seurat Object.

<input type="checkbox"> For the `Plot_type_selector` dropdown menu, select `VlnPlot`.

<input type="checkbox"> For the `Plot_type_selector` dropdown menu, select `VlnPlot`.

<input type="checkbox"> For the `Features` field type in `nCount_RNA`.

<input type="checkbox"> Scroll down to `Log` and toggle it to yes.

<input type="checkbox"> Click the Run Tool button.

You should see a green message about the job being in the queue and a new block will show up in the history that says plot. It will take some time for the plot to be generated.  


<input type="checkbox"> Once the new block turns green click on the eye button to see the plot.

<p align="center" >
    <img width="50%" src="resources/images/plot_view_singlecell.png" class="img_shad">
</p>


#### QC: Visualize Features

#### Finding our filtering parameters

#### Filter Cells

#### Normalize Data

#### Find Variable Genes

#### Scale Data

### Dimensionality Reduction

#### Run PCA

#### Run UMAP

Note here this walkthrough diverges from the Galaxy Training tutorial, rearranging the dimensionality reduction steps.

Change the input such that for this tutorial it should be the PCA Processed Seurat Object (instead of the Preprocessed Seurat Object with Clusters)

Keep the Dims 1:15 argument.

The Reduction argument autofilled to be PCA and I kept it.


#### Plot Gapdh

Use the output of the previous step "Seurat UMAP on data X: Seurat RDS" as the input for this one.

Keep the Plot_type_selector as FeaturePlot and Features as Gapdh as described in the Galaxy Training Tutorial.

#### Plot Il2ra

Use the output of the Run UMAP step "Seurat UMAP on data X: Seurat RDS" as the input for this one.

Keep the Plot_type_selector as FeaturePlot and Features as Il2ra as described in the Galaxy Training Tutorial.

NOTE: If you plot UMAP at this stage, do not use "Group by: RNA_nn_res.0.5" argument

#### Find Neighbors

Input is Y: Seurat UMAP on data X: Seurat RDS (the output of the Run UMAP step "Seurat UMAP on data X: Seurat RDS")

“Reduction”: PCA
“Dimensions”: 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15
“Assay”: RNA
--> Z: Seurat FindNeighbours on data Y: Seurat RDS

#### Find Clusters

In “Advanced Options “
“Resolution”: 0.5

Z:Seurat FindNeighbours on data Y: Seurat RDS
--> A: Seurat FindClusters on data Z: Seurat RDS

#### Plot UMAP

Group by: RNA_nn_res.0.5

A: Seurat FindClusters on data Z: Seurat RDS
--> B: Seurat DimPlot on data A: png plot

## Other Resources



- [Further information on Seurat and its manual](https://satijalab.org/seurat/)

- [UMAP discussion](https://alleninstitute.org/resource/what-is-a-umap/)
