---
title: "Single Cell Transcriptomics Activity"
output: html_document
---

Note the galaxy link they should use: https://usegalaxy.org/join-training/itn-at-moffitt-2025/

## Tutorial Steps

### Get Data onto Galaxy and generate a Seurat object

#### EBI Data Retrieval

#### Seurat Read10X

### QC and further processing of the Seurat object

#### QC: Visualize Counts

#### QC: Visualize Features

#### Finding our filtering parameters

#### Filter Cells

#### Normalize Data

#### Find Variable Genes

#### Scale Data

### Dimensionality Reduction

#### Run PCA

#### Run UMAP

Note here this walkthrough diverges from the Galaxy Training tutorial, rearranging the dimensionality reduction steps.

Change the input such that for this tutorial it should be the PCA Processed Seurat Object (instead of the Preprocessed Seurat Object with Clusters)

Keep the Dims 1:15 argument.

The Reduction argument autofilled to be PCA and I kept it.


#### Plot Gapdh

Use the output of the previous step "Seurat UMAP on data X: Seurat RDS" as the input for this one.

Keep the Plot_type_selector as FeaturePlot and Features as Gapdh as described in the Galaxy Training Tutorial.

#### Plot Il2ra

Use the output of the Run UMAP step "Seurat UMAP on data X: Seurat RDS" as the input for this one.

Keep the Plot_type_selector as FeaturePlot and Features as Il2ra as described in the Galaxy Training Tutorial.

NOTE: If you plot UMAP at this stage, do not use "Group by: RNA_nn_res.0.5" argument

#### Find Neighbors

Input is Y: Seurat UMAP on data X: Seurat RDS (the output of the Run UMAP step "Seurat UMAP on data X: Seurat RDS")

“Reduction”: PCA
“Dimensions”: 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15
“Assay”: RNA
--> Z: Seurat FindNeighbours on data Y: Seurat RDS

#### Find Clusters

In “Advanced Options “
“Resolution”: 0.5

Z:Seurat FindNeighbours on data Y: Seurat RDS
--> A: Seurat FindClusters on data Z: Seurat RDS

#### Plot UMAP

Group by: RNA_nn_res.0.5

A: Seurat FindClusters on data Z: Seurat RDS
--> B: Seurat DimPlot on data A: png plot

## Other Resources

- [Full Galaxy Training Tutorial "Filter, plot, and explore single cell RNA-seq data with Seurat"](https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html) from which this tutorial was adapted. Changes include
  - several subsections within QC that are omitted
  - rearranging of the Dimensionality Reduction section
  - Stopping before the Differential Expression Testing: Finding Markers section

- [Further information on Seurat and its manual](https://satijalab.org/seurat/)

- [UMAP discussion](https://alleninstitute.org/resource/what-is-a-umap/)
