<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Single Cell Transcriptomics Activity</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />
<link rel="shortcut icon" href="resources/images/favicon.ico" />
 <!--- go to https://favicon.io/favicon-converter/ to upload an image to make a new favicon.io. You will need to replace the current favicon.io image with the one in the downloaded directory from the website. The current image is in the resources/images/ directory --->

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Moffitt Workshops 2025</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Computing and Ethics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="computing_workshop.html">Workshop Material</a>
    </li>
    <li>
      <a href="computing_activity.html">Activity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    scRNA-seq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="scRNAseq_workshop.html">Workshop Material</a>
    </li>
    <li>
      <a href="sc-gene-expression-workshop.html">Activity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Spatial Transcriptomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="spatialTranscriptomics_workshop.html">Workshop Material</a>
    </li>
    <li>
      <a href="st-activity-instructions.html">Activity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Single Cell Protein Imaging
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="scProteinImaging_workshop.html">Workshop Material</a>
    </li>
    <li>
      <a href="sc-proteomics.html">Activity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Machine Learning
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ml_workshop.html">Workshop Material</a>
    </li>
    <li>
      <a href="machine-learning.html">Activity</a>
    </li>
  </ul>
</li>
<li>
  <a href="https://www.itcrtraining.org/home">ITN Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Single Cell Transcriptomics Activity</h1>
<h3 class="subtitle">Activity</h3>

</div>


<div id="tutorial-steps" class="section level2">
<h2>Tutorial Steps</h2>
<p>We will be working through a modified/truncated version of the <a
href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">Galaxy
Training Tutorial “Filter, plot, and explore single cell RNA-seq data
with Seurat” Tutorial</a>.</p>
<p>Changes include:</p>
<ul>
<li>several subsections within QC will be omitted</li>
<li>we rearranged the Dimensionality Reduction section</li>
<li>we will stop before the Differential Expression Testing: Finding
Markers section</li>
</ul>
<div class="warning">
<p>You may want to view the whole Galaxy Training Tutorial later for
further steps or explanation, but we would recommend not following along
with it during this activity considering that we have abbreviated and
rearranged a portion of it.</p>
</div>
</div>
<div id="setting-up-galaxy" class="section level2">
<h2>Setting up Galaxy</h2>
<div id="open-galaxy" class="section level3">
<h3>Open Galaxy</h3>
<p><input type="checkbox"> Please <a
href="https://usegalaxy.org/join-training/itn-at-moffitt-2025/">use this
link to access the Galaxy platform specifically for our
activity</a>.<br/></p>
<p>Note: this link will only be active for this workshop, but you can
continue to use Galaxy resources for free by visiting <a
href="https://usegalaxy.org/">usegalaxy.org</a>.</p>
</div>
<div id="set-up-galaxys-history-pane" class="section level3">
<h3>Set up Galaxy’s history pane</h3>
<p><input type="checkbox"> If you have files in your history already,
use the <code>+</code> button on the top right of the history pane to
<code>Create new history</code>. <br/> <input type="checkbox"> Click the
pencil button next to “Unnamed history”. Fill in the name with something
descriptive/appropriate. Click “Save”.</p>
<p align="center">
<img width="40%" src="resources/images/hist_singlecell.png" class="img_shad">
</p>
<p>Now we are ready to start the tutorial!</p>
</div>
</div>
<div id="get-data-onto-galaxy-and-generate-a-seurat-object"
class="section level2">
<h2>Get Data onto Galaxy and generate a Seurat object</h2>
<p><input type="checkbox"> Click on the “Get Data onto Galaxy and
generate a Seurat object” link to get started in the tutorial.</p>
<div id="ebi-data-retrieval" class="section level3">
<h3>EBI Data Retrieval</h3>
<p><input type="checkbox"> Copy and paste
<code>EBI SCXA Data Retrieval</code> into the tool search.</p>
<p align="center">
<img width="40%" src="resources/images/data_singlecell.png" class="img_shad">
</p>
<p><input type="checkbox"> Select
<code>EBI SCXA DATa Retrieval</code>.</p>
<p><input type="checkbox"> Copy paste <code>E-MTAB-6945</code> into the
<code>SC-Atlas experiment accession</code> field</p>
<p><input type="checkbox"> Make sure the
<code>“Choose the type of matrix to download”</code> field says
<code>Raw filtered counts</code>.</p>
<p><input type="checkbox"> Click the Run tool button.</p>
<p align="center">
<img width="50%" src="resources/images/EBI_singlecell.png" class="img_shad">
</p>
<p>You will see your history pane show some new activity, with
rectangles showing the data file upload first in gray (to show that the
job has been requested), then yellow (to show that Galaxy is working on
it), and finally green (to show that the job is complete).</p>
<p>Note: do not worry about the numbers next to each step - depending on
how many times you run requests you will have different numbers.</p>
<p align="center">
<img width="50%" src="resources/images/gal_data_singlecell.png" class="img_shad">
</p>
<p>You have now imported the <code>exp_design.tsv</code>,
<code>barcodes.tsv</code>, <code>genes.tsv</code>,
<code>matrix.mtx</code> and files into Galaxy! Note that each file
starts with <code>EBI SCXA Data Retrieval on E-MTAB-6945</code> to
indicate how we got them.</p>
</div>
<div id="generating-a-seurat-read10x-object" class="section level3">
<h3>Generating a Seurat Read10X Object</h3>
<p>Now we need to get our data into a format that the Seurat tools will
recognize.</p>
<p><input type="checkbox"> Copy paste <code>Seurat Read10x</code> into
the tools search.</p>
<p align="center">
<img width="50%" src="resources/images/10x_singlecell.png" class="img_shad">
</p>
<p><input type="checkbox"> Select <code>Seurat Read10x</code>.</p>
<p><input type="checkbox"> Select the pull down menu for the
<code>Expression matrix in sparse matrix format (.mtx)</code> and select
the <code>matrix.mtx</code> file.</p>
<p align="center">
<img width="50%" src="resources/images/select_file_singlecell.png" class="img_shad">
</p>
<p>Continue with the rest of the files to select the appropriate file
for each field. Remember that each file starts with
<code>EBI SCXA Data Retrieval on E-MTAB-6945</code>.</p>
<p><input type="checkbox"> Select the pull down menu for the
<code>Gene table</code> and select the <code>genes.tsv</code> file.</p>
<p><input type="checkbox"> Select the pull down menu for the
<code>Barcode/cell table</code> and select the <code>barcodes.tsv</code>
file.</p>
<p><input type="checkbox"> Select the pull down menu for the
<code>Cell Metadata</code> and select the <code>exp_design.tsv</code>
file.</p>
<p><input type="checkbox"> Type <code>5</code> into the
<code>Minimum cells to include features</code>.</p>
<p><input type="checkbox"> Ensure that the
<code>Choose format of the output</code> field says
<code>RDS with a Seurat object</code>.</p>
<p><input type="checkbox"> Rename the output to
<code>Initial Seurat Object</code>.</p>
<p align="center">
<img width="50%" src="resources/images/seurat_singlecell.png" class="img_shad">
</p>
<p><input type="checkbox"> Click <code>Run Tool</code> Button.</p>
<p>You should see a green message indicating that the job is now in the
queue.</p>
<p><input type="checkbox"> Click on the pencil button of the new output
that shows up in the history pane.</p>
<p align="center">
<img width="50%" src="resources/images/rename_singlecell.png" class="img_shad">
</p>
<p><input type="checkbox"> Replace the name of the object to
<code>Initial Seurat Object</code> and press save.</p>
<p align="center">
<img width="50%" src="resources/images/rename2_singlecell.png" class="img_shad">
</p>
<p>Now we have a Seurat object so we can start using Seurat to work with
our data!</p>
</div>
</div>
<div id="qc-and-further-processing-of-the-seurat-object"
class="section level2">
<h2>QC and further processing of the Seurat object</h2>
<p>Our first step is to do some Quality Control checks aka QC to
evaluate our data.</p>
<div id="qc-visualize-counts" class="section level3">
<h3>QC: Visualize Counts</h3>
<p>There will be a number of “cells” that are actually just empty
droplets or low-quality. There will also be genes that could be
sequencing artifacts or that appear with such low frequency that
statistical tools will fail to accurately analyze them.</p>
<p>This background noise of both cells and genes not only makes it
harder to distinguish real biological information from artifacts, but
also makes it computationally demanding to analyze.</p>
<p>We want to filter our cells, but first we need to know what our data
looks like to make appropriate filters.</p>
<p>We will not have time to go through all of the possible plots one
might use, but we will do some evaluations. Remember that you can always
do the full tutorial in your own time.</p>
<p><input type="checkbox"> Search for the <code>Plot with Seurat</code>
tool in the tool search.</p>
<p><input type="checkbox"> Select <code>**Plot** with Seurat</code>.</p>
<p align="center">
<img width="50%" src="resources/images/plot_tool_singlecell.png" class="img_shad">
</p>
<p><input type="checkbox"> Ensure that the RDS file field shows our
Initial Seurat Object.</p>
<p><input type="checkbox"> For the <code>Plot_type_selector</code>
dropdown menu, select <code>VlnPlot</code>.</p>
<p>This stands for violin plot.</p>
<p><input type="checkbox"> For the <code>Features</code> field type in
<code>nCount_RNA</code>.</p>
<p><input type="checkbox"> Scroll down to <code>Log</code> and toggle it
to yes.</p>
<p><input type="checkbox"> Click the Run Tool button.</p>
<p>You should see a green message about the job being in the queue and a
new block will show up in the history that says plot. It will take some
time for the plot to be generated.</p>
<p><input type="checkbox"> Once the new block turns green click on the
eye button to see the plot.</p>
<p align="center">
<img width="50%" src="resources/images/plot_view_singlecell.png" class="img_shad">
</p>
<p>This plot shows us the spread of cells in our data containing a given
number of counts (or transcripts) observed in a given cell. We can use
this plot, and others like it in a moment, to help filter out the
uninformative cells.</p>
<p align="center">
<img width="50%" src="resources/images/plot1_singlecell.png" class="img_shad">
</p>
<p>We see a severe drop off in the number of cells that contain fewer
than 500 and more than 10,000 transcripts. These will be our nCount
thresholds that we filter based on.</p>
</div>
<div id="qc-visualize-features" class="section level3">
<h3>QC: Visualize Features</h3>
<p>In a similar fashion we can visualize the spread of cells in our data
expressing a given number of features (or genes).</p>
<p><input type="checkbox"> Search for the <code>Plot with Seurat</code>
tool in the tool search again.</p>
<p><input type="checkbox"> Select <code>**Plot** with Seurat</code>.</p>
<p><input type="checkbox"> Ensure that the RDS file field shows our
Initial Seurat Object.</p>
<p><input type="checkbox"> For the <code>Plot_type_selector</code>
dropdown menu, select <code>VlnPlot</code>.</p>
<p><input type="checkbox"> For the <code>Features</code> field type in
<code>nFeature_RNA</code>.</p>
<p><input type="checkbox"> Scroll down to <code>Log</code> and toggle it
to yes.</p>
<p><input type="checkbox"> Click the Run Tool button.</p>
<p>Now, we could pick filtering thresholds based on these plots, and in
a typical pipeline we would also plot the proportion of features that
map to the mitochondrial genome and to other QC checks but we will stop
here for now.</p>
<p align="center">
<img width="50%" src="resources/images/plot2_singlecell.png" class="img_shad">
</p>
<p>In this plot, we can see that very few cells in the dataset contain
fewer than ~500 genes. Biologically, this makes sense, and the cells
appear to be outliers in the data. As such, we will set our lower
threshold of genes (nFeature_RNA) at 500.</p>
</div>
<div id="filter-cells" class="section level3">
<h3>Filter Cells</h3>
<p>Recall that from our first plot we saw a severe drop off in the
number of cells that contain fewer than 500 and more than 10,000
transcripts. These will be our nCount thresholds that we filter based
on. and that our lower threshold of genes would be 500 from our second
plot.</p>
<p>Will will now apply these thresholds to filter our data.</p>
<p><input type="checkbox"> Search for the
<code>Seurat FilterCells</code> tool in the tool search.</p>
<p><input type="checkbox"> Select <code>Seurat FilterCells</code>.</p>
<p><input type="checkbox"> Ensure that the RDS file field shows our
Initial Seurat Object.</p>
<p><input type="checkbox"> In Subsets used to filter cells”,
<code>Name of Parameter to filter on</code> field, select
<code>nCount_RNA</code> from the dropdown menu.</p>
<p><input type="checkbox"> Select 500.0 for the Minimum value.</p>
<p><input type="checkbox"> Select 10000.0 for the Maximum value.</p>
<p><input type="checkbox"> Click on the button that says
<code>Insert Subsets used to filter cells</code></p>
<p align="center">
<img width="50%" src="resources/images/subset_button_singlecell.png" class="img_shad">
</p>
<p><input type="checkbox"> For the
<code>Name of Parameter to filter on</code> field, select
<code>nFeature_RNA</code> from the dropdown menu.</p>
<p><input type="checkbox"> Select 500.0 for the Minimum value.</p>
<p><input type="checkbox"> Select 1000000000.0 for the Maximum
value.</p>
<p><input type="checkbox"> Click the Run Tool button.</p>
<p>You will see a green message a new block in the history that says
“Seurat Filter cells on data”, after a few moments it will turn
green.</p>
<p><input type="checkbox"> Rename the output object to be
<code>Filtered Seurat Object</code> by clicking on the pencil button and
replacing the name and saving the changes.</p>
</div>
<div id="normalize-data" class="section level3">
<h3>Normalize Data</h3>
<p>Now we will normalize our data. This helps reduce the differences
between gene counts by fitting total counts across cells in our data to
be comparable to one another. SCTransform regularizes the gene
expression profiles via a negative binomial regression while also
controlling for overfitting of the data.</p>
<p><input type="checkbox"> Search for the
<code>Seurat NormaliseData</code> tool in the tool search.</p>
<p><input type="checkbox"> Select <code>Seurat NormaliseData</code>.</p>
<p><input type="checkbox"> Ensure that the RDS file field shows our
<em>Filtered</em> Seurat Object.</p>
<p><input type="checkbox"> For the <code>Normalization Method</code>
ensure that the drop down menu shows <code>log Normalise</code>.</p>
<p><input type="checkbox"> Click the Run Tool button.</p>
<p><input type="checkbox"> Once the new SeuratNormalise Data object
turns green in the history pane, rename it to be
<code>Normalised Seurat Object</code> by clicking on the pencil button
and replacing the name and saving the changes.</p>
</div>
<div id="find-variable-genes" class="section level3">
<h3>Find Variable Genes</h3>
<p>The datasets have loads of genes, but not all of them vary in
expression from cell to cell. For instance, housekeeping genes are
defined as not changing much from cell to cell, so we could remove these
from our data to simplify our analysis.</p>
<p>The find variable genes, this step flags genes that do vary across
cells to expedite future analyses and ensure that we, and Seurat, don’t
waste time looking for meaningful differences where they don’t
exist.</p>
<p><input type="checkbox"> Search for the
<code>Seurat FindVariableGenes</code> tool in the tool search.</p>
<p><input type="checkbox"> Select
<code>Seurat FindVariableGenes</code>.</p>
<p><input type="checkbox"> Ensure that the <code>RDS file</code> field
shows our <em>Normalised</em> Seurat Object.</p>
<p><input type="checkbox"> Ensure that the
<code>Choose the format of the output</code> field shows RDS with a
Seurat Object.</p>
<p><input type="checkbox"> Click the Run Tool button.</p>
<p>This will result in the creation of two new objects:</p>
<ol style="list-style-type: decimal">
<li>A new Seurat object with variable features identified and
flagged</li>
<li>A tabular file with a list of these variable genes.</li>
</ol>
<p>This gene list may be used as a sneak peak into understanding what
the dataset will look like! We can begin to understand which genes are
going to be driving downstream clustering of our cells and maybe even
make some decisions about whether we are happy with our filtering based
on this list.</p>
<p><input type="checkbox"> Rename the new Seurat object called
<code>Seurat FindVariableGenes on Data some number:Seurat RDS</code> to
be <code>Normalised Seurat Object with Variable Features</code> by
clicking on the pencil button, replacing the name and clicking save.</p>
</div>
<div id="scale-data" class="section level3">
<h3>Scale Data</h3>
<p>Now we will scale the data.</p>
<p>What is scaling?</p>
<p>This is an important step to set up our data for further
dimensionality reduction. It will transform the dataset such that all
genes have the same variance and a zero mean. It helps negate sequencing
depth differences between samples, since the gene levels across the
cells become comparable.</p>
<p>Note: that the differences from scaling etc. are not the values you
have at the end - i.e. if your cell has average GAPDH levels, it will
not appear as a “0” when you calculate gene differences between
clusters.</p>
<p><input type="checkbox"> Search for the <code>Seurat ScaleData</code>
tool in the tool search.</p>
<p><input type="checkbox"> Select <code>Seurat ScaleData</code>.</p>
<p><input type="checkbox"> Ensure that the <code>RDS file</code> field
shows our object called Normalised Seurat Object with Variable
Features.</p>
<p><input type="checkbox"> Ensure that the
<code>Choose the format of the output</code> field shows RDS with a
Seurat Object.</p>
<p><input type="checkbox"> For the <code>Genes to use</code> dropdown
menu select
<code>Seurat FindVariableGenes on data 12: Variable genes **tabular file**</code>.
This is the gene list we just made in the last step.</p>
<p><input type="checkbox"> For <code>Vars to regress</code> ensure that
it says <code>nCount_RNA</code>.</p>
<p>This function allow us to mitigate the effects of confounding factors
in our dataset. In true research practice, one might regress out
multiple variables including but not limited to percent mitochondria,
cell cycle scoring, and feature count. As currently written, this tool
only allows us to regress out a single variable.</p>
<p><input type="checkbox"> For <code>Statistical Model</code> ensure
that it says <code>Linear model</code>.</p>
<p><input type="checkbox"> Click the Run Tool button.</p>
<p>This step can take a few moments.</p>
<p><input type="checkbox"> Once the new object turns green, rename it to
be <code>Scaled,Preprocessed Seurat Object</code> by clicking on the
pencil button, replacing the name and clicking save.</p>
<p>Nice! We now have a preprocessed Seurat object!</p>
</div>
</div>
<div id="dimensionality-reduction" class="section level2">
<h2>Dimensionality Reduction</h2>
<p>Although we’ve made our expression values comparable to one another
and our overall dataset less computationally demanding, we still have
way too many dimensions (n cells x n genes!).</p>
<p>Transcript changes are not usually singular–which is to say, genes
function and exist in pathways and groups. It would be easier to analyze
our data if we could group these differences. To address this we will
run principal component analysis (PCA).</p>
<div id="run-pca" class="section level3">
<h3>Run PCA</h3>
<p>Principal components (PCs) are calculated from highly dimensional
data to find the most representative spread in the dataset. So in our
highly variable gene dimensions, there will be one line (axis) that
yields the most spread and variation across the cells. That will be our
first principal component.</p>
<p>We can calculate the first handful of principal components in our
data to drastically reduce the number of dimensions.</p>
<p><input type="checkbox"> Search for the <code>Seurat RunPCA</code>
tool in the tool search.</p>
<p><input type="checkbox"> Select <code>Seurat RunPCA</code>.</p>
<p><input type="checkbox"> Ensure that the <code>RDS file</code> field
shows our object called Scaled, Preprocessed Seurat Object.</p>
<p><input type="checkbox"> Ensure that the
<code>Choose the format of the output</code> field shows RDS with a
Seurat Object.</p>
<p><input type="checkbox"> For the dropdown menu in the
<code>Genes to Scale</code> section, select the variable genes tabular
file.</p>
<p align="center">
<img width="50%" src="resources/images/pca_setup_singlecell.png" class="img_shad">
</p>
<p><input type="checkbox"> Click the Run Tool button.</p>
<p>You will see 4 new objects: <code>Seurat Std dev</code>,
<code>Seurat Loadings</code>, <code>Seurat Embeddings</code>, and
<code>Seurat RDS</code> (the object type we have mostly been using).</p>
<p><input type="checkbox"> Once the new object turns green, rename the
one that says <code>Seurat RDS</code> to be
<code>PCA Processed Seurat Object</code> by clicking on the pencil
button, replacing the name and clicking save.</p>
<p>So what are all these new objects?</p>
<ul>
<li>Seurat RDS - includes all of the following PCA metadata</li>
<li>Seurat Embeddings - Principal component values for each of the cells
in your dataset</li>
<li>Seurat Loadings - Principal component values for each of the genes
in your dataset</li>
<li>Seurat Std dev - Standard deviations of each principal component
coordinates</li>
</ul>
</div>
<div id="run-umap" class="section level3">
<h3>Run UMAP</h3>
<p>Next we will use UMAP, the most recently developed, and most widely
used dimensionality reduction method for visualization of principal
component data. It has been optimized since tSNE to better preserve
global structure and is less computationally demanding.</p>
<p>Note that this is where the walkthrough diverges from the Galaxy
Training tutorial, rearranging the dimensionality reduction steps.</p>
<p><input type="checkbox"> Search for the <code>Seurat UMAP</code> tool
in the tool search.</p>
<p><input type="checkbox"> Select <code>Seurat UMAP</code>.</p>
<p><input type="checkbox"> Ensure that the <code>RDS file</code> field
shows our object called PCA Processed Seurat Object.</p>
<p><input type="checkbox"> Ensure that the
<code>Choose the format of the output</code> field shows RDS with a
Seurat Object.</p>
<p><input type="checkbox"> For the <code>Dims</code> type
<code>1:15</code>.</p>
<p><input type="checkbox"> Click the Run Tool button.</p>
</div>
<div id="plot-gapdh" class="section level3">
<h3>Plot Gapdh</h3>
<p><input type="checkbox"> Search for the <code>Plot with Seurat</code>
tool in the tool search.</p>
<p><input type="checkbox"> Select <code>Plot with Seurat</code>.</p>
<p><input type="checkbox"> Ensure that the <code>RDS file</code> field
shows “Seurat UMAP on data X: Seurat RDS” as the input for this step
(this file is the output from the previous step).</p>
<p><input type="checkbox"> Confirm that the
<code>Plot_type_selector</code> option is “FeaturePlot”.</p>
<p><input type="checkbox"> For the <code>Features</code> option, type
<code>Gapdh</code>.</p>
<p><input type="checkbox"> Click the Run Tool button.</p>
<p><input type="checkbox"> View the resulting plot by clicking the eye
icon (<img src="resources/images/eye.png" id="id" class="class"
width="25" height="20" alt="eye button image" />) next to the output
<code>.png</code> file in the history.</p>
</div>
<div id="plot-il2ra" class="section level3">
<h3>Plot Il2ra</h3>
<p><input type="checkbox"> Search for the <code>Plot with Seurat</code>
tool in the tool search.</p>
<p><input type="checkbox"> Select <code>Plot with Seurat</code>.</p>
<p><input type="checkbox"> Ensure that the <code>RDS file</code> field
shows “Seurat UMAP on data X: Seurat RDS” as the input for this step
(this file is the output from the Run UMAP step).</p>
<p><input type="checkbox"> Confirm that the
<code>Plot_type_selector</code> option is “FeaturePlot”.</p>
<p><input type="checkbox"> For the <code>Features</code> option, type
<code>Il2ra</code>.</p>
<p><input type="checkbox"> Click the Run Tool button.</p>
<p><input type="checkbox"> View the resulting plot by clicking the eye
icon (<img src="resources/images/eye.png" id="id" class="class"
width="25" height="20" alt="eye button image" />) next to the output
<code>.png</code> file in the history.</p>
<details>
<summary>
Can I plot UMAP clusters at this stage?
</summary>
If you wanted to plot a UMAP projection at this stage using the
<code>Plot with Seurat</code> tool and <code>Plot_type_selector</code>
<code>DimPlot</code> option (as described in the Galaxy Training
tutorial), do not use “Group by: RNA_nn_res.0.5” argument. Because the
Find Neighbors and Find Clusters steps haven’t been used yet, this
option isn’t relevant yet, and you won’t be able to visualize clusters.
</details>
</div>
<div id="find-neighbors" class="section level3">
<h3>Find Neighbors</h3>
<p><input type="checkbox"> Search for the
<code>Seurat FindNeighbours</code> tool in the tool search.</p>
<p><input type="checkbox"> Ensure that the <code>RDS file</code> field
shows “Seurat UMAP on data X: Seurat RDS” as the input for this step
(this file is the output from the Run UMAP step).</p>
<p><input type="checkbox"> For the <code>Reduction</code> option, type
<code>pca</code>.</p>
<p><input type="checkbox"> For the <code>Dimensions</code> option, copy
and paste <code>1,2,3,4,5,6,7,8,9,10,11,12,13,14,15</code>.</p>
<p><input type="checkbox"> For the <code>Assay</code> option, type
<code>RNA</code>.</p>
<p><input type="checkbox"> Click the Run Tool button.</p>
<p>The output of this will be “Seurat FindNeighbours on data Y: Seurat
RDS” (and a corresponding CSV file)</p>
</div>
<div id="find-clusters" class="section level3">
<h3>Find Clusters</h3>
<p><input type="checkbox"> Search for the
<code>Seurat FindClusters</code> tool in the tool search.</p>
<p><input type="checkbox"> Ensure that the <code>RDS file</code> field
shows “Seurat FindNeighbours on data Y: Seurat RDS” as the input for
this step.</p>
<p><input type="checkbox"> In the <code>Advanced Options</code> section,
for the <code>Resolution</code> option, type <code>0.5</code></p>
<p><input type="checkbox"> Click the Run Tool button.</p>
<p>The output of this will be “Seurat FindClusters on data Z: Seurat
RDS” (and a corresponding CSV file)</p>
</div>
<div id="plot-umap" class="section level3">
<h3>Plot UMAP</h3>
<p><input type="checkbox"> Search for the <code>Plot with Seurat</code>
tool in the tool search.</p>
<p><input type="checkbox"> Ensure that the <code>RDS file</code> field
shows “Seurat FindClusters on data Z: Seurat RDS” as the input for this
step.</p>
<p><input type="checkbox"> Select “DimPlot” for the
<code>Plot_type_selector</code> option.</p>
<p><input type="checkbox"> For the <code>Group by</code> option, type
<code>RNA_nn_res.0.5</code></p>
<p><input type="checkbox"> Click the Run Tool button.</p>
<p><input type="checkbox"> View the resulting plot by clicking the eye
icon (<img src="resources/images/eye.png" id="id" class="class"
width="25" height="20" alt="eye button image" />) next to the output
<code>.png</code> file in the history.</p>
</div>
</div>
<div id="other-resources" class="section level2">
<h2>Other Resources</h2>
<ul>
<li><p><a href="https://satijalab.org/seurat/">Further information on
Seurat and its manual</a></p></li>
<li><p><a
href="https://alleninstitute.org/resource/what-is-a-umap/">UMAP
discussion</a></p></li>
</ul>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
